ml/jml/decision_tree.h

```diff
@@ -120,7 +120,8 @@ class Decision_Tree : public Classifier_Impl {
                            const ML::Label & label,
                            double weight,
                            const Tree::Ptr & ptr,
-                           const Tree::Node * parent) const;
+                           const Tree::Node * parent,
+                           std::function<float (const Tree::Base&)>& get_pred) const;
```

Comment:
should be a const reference to the function, or ideally a simple function pointer